db.go

```diff
@@ -571,3 +572,92 @@ func (db *DB) rewriteValidRecords(walFile *wal.WAL, validRecords []*ValueLogReco
 	}
 	return db.index.PutBatch(positions, matchKeys...)
 }
+
+func (db *DB) NewIterator(options IteratorOptions) (*MergeIterator, error) {
+	if db.options.IndexType != BTree {
+		return nil, ErrDBIteratorUnsupportedType
+	}
+	db.mu.Lock()
+	defer func() {
+		if r := recover(); r != nil {
+			db.mu.Unlock()
+		}
+	}()
+	itrs := make([]*SingleIter, 0, db.options.PartitionNum+len(db.immuMems)+1)
```

Comment:
What if the user set `partition num = 3` at first, then restart db with `partition num = 5`, will this be a problem?

Reply:
No problem. If the option remains unchanged, `db.options.PartitionNum+len(db.immuMems)+1 `represents all the necessary memory we need, and we only need to allocate memory once. Additionally, in the initialization of itrs, only the` append `method was used, and it will automatically expand if the capacity is insufficient.